library/datasort.h

```diff
@@ -81,8 +81,10 @@ struct datasort_cfg {
 	struct list_head		sorted_chunks;
 	/* Result of mergesort */
 	struct datasort_chunk		*result;
-	/* Datasort directory */
-	char				*dir;
+	/* Datasort directory - chunks are merged into single sorted blob here */
+	char				*datasort_dir;
+	/* Chunks directory - if not defined, then chunks are stored in datasort_dir */
+	char				*chunks_dir;
```

Comment:
Ok, this makes sense.

Please rename `datasort_dir` to `data_dir` or just `dir` like it was before, since there is no sorting made in this directory.
Also setup chunk directory in init function (to default value and change it if there is appropriate config option) and use it everywhere without those `if` and flags